lib/CAPI/RTL/RTLDialect.cpp

```diff
@@ -0,0 +1,23 @@
+//===- RTLDialect.cpp - C Interface for the RTL Dialect -------------------===//
+//
+//  Implements a C Interface for the RTL Dialect
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt-c/RTLDialect.h"
+#include "circt/Dialect/RTL/Ops.h"
+#include "mlir-c/IR.h"
+#include "mlir/CAPI/IR.h"
+#include "mlir/CAPI/Support.h"
+
+void mlirContextRegisterRTLDialect(MlirContext context) {
```

Comment:
Am I missing something or should these symbols be `extern "C"`?

Reply:
I mirrored the structure of StandardDialect.cpp: https://github.com/llvm/llvm-project/blob/64c0c9f01511dc300b29e7a20a13958c5932e314/mlir/lib/CAPI/Standard/StandardDialect.cpp
They are declared as `extern "C"` in the header: https://github.com/llvm/circt/blob/4b01c0ab7594ef32955f49f9f7dc44e1e96881c2/include/circt-c/RTLDialect.h#L17